File: Javascript/backend_nest/src/services/UserService.ts
import { User } from 'src/domain/entities/User';
import { UserRepository } from 'src/domain/repositories/UserRepository';
import { IUserService } from 'src/domain/services/UserService';

export class UserService implements IUserService {
  constructor(private readonly repository: UserRepository) {}

  async createUser(user: User): Promise<User> {
    return this.repository.add(user);
  }

  async getUserById(userId: number): Promise<User | null> {
    return this.repository.findOne(userId);
  }

  async listAllUsers(): Promise<User[]> {
    return this.repository.findMany();
  }

  async updateUser(userId: number, updatedUser: User): Promise<User | null> {
    return this.repository.update(userId, updatedUser);
  }

  async deleteUser(userId: number): Promise<boolean> {
    return this.repository.delete(userId);
  }
}

File: Python/backend_flask_archi_feature/user/user_service.py
from typing import Optional, List
from user.user import User
from user.user_repository import UserRepository

class UserService:
    def __init__(self, repository: UserRepository):
        self.repository = repository

    def get_user_by_id(self, user_id: int) -> Optional[User]:
        return self.repository.get_user(user_id)

    def list_all_users(self) -> List[User]:
        return self.repository.list_users()

    def update_user(self, user_id: int, updated_user: User) -> Optional[User]:
        return self.repository.update_user(user_id, updated_user)

    def delete_user(self, user_id: int) -> bool:
        return self.repository.delete_user(user_id)